server/src/main/java/org/opensearch/index/mapper/BooleanFieldMapper.java

```diff
@@ -257,9 +278,72 @@ public DocValueFormat docValueFormat(@Nullable String format, ZoneId timeZone) {
             return DocValueFormat.BOOLEAN;
         }
 
+        @Override
+        public Query termQuery(Object value, QueryShardContext context) {
+            failIfNotIndexedAndNoDocValues();
+            Query query = new TermQuery(new Term(name(), indexedValueForSearch(value)));
+            if (boost() != 1f) {
+                query = new BoostQuery(query, boost());
+            }
+            if (isSearchable() && hasDocValues()) {
+                return new IndexOrDocValuesQuery(
+                    query,
+                    SortedNumericDocValuesField.newSlowExactQuery(name(), Values.TRUE.bytesEquals(indexedValueForSearch(value)) ? 1 : 0)
+                );
+            }
+            if (hasDocValues()) {
+                return SortedNumericDocValuesField.newSlowExactQuery(name(), Values.TRUE.bytesEquals(indexedValueForSearch(value)) ? 1 : 0);
+            }
+            return query;
+        }
+
+        @Override
+        public Query termsQuery(List<?> values, QueryShardContext context) {
+            failIfNotIndexedAndNoDocValues();
+            // if we do not get either True or False, we return no docs
+            if (!(values.contains(ExpandedValues.TRUE)) && !(values.contains(ExpandedValues.FALSE))) {
+                return new MatchNoDocsQuery("Values do not contain True or False");
+            }
+            // if we have either True or False, we delegate to termQuery
+            if ((values.contains(ExpandedValues.TRUE) && !(values.contains(ExpandedValues.FALSE)))
+                || (values.contains(ExpandedValues.FALSE) && !values.contains(ExpandedValues.TRUE))) {
+                return termQuery(values.contains(ExpandedValues.TRUE) ? ExpandedValues.TRUE : ExpandedValues.FALSE, context);
+            }
+            // if we have both True and False, we acknowledge that the field exists with a value
+            return new FieldExistsQuery(name());
+        }
```

Comment:
Given that there are several possible values that can map to true or false, I would lean on `indexedValueForSearch` to do that mapping.

Maybe:
```
        public Query termsQuery(List<?> values, QueryShardContext context) {
            boolean seenTrue = false;
            boolean seenFalse = false;
            for (Object value : values) {
                if (Values.TRUE.equals(indexedValueForSearch(value))) {
                    seenTrue = true;
                } else {
                    // The only other thing indexedValueForSearch could return without exception is false.
                    seenFalse = true;
                }
            }
            if (seenTrue) {
                if (seenFalse) {
                    return new MatchAllDocsQuery();
                }
                return termQuery(Values.TRUE, context);
            }
            if (seenFalse) {
                return termQuery(Values.FALSE, context);
            }
            return new MatchNoDocsQuery("Values did not contain true or false");
        }
```

Also, probably a good idea to add some tests covering string values and boolean values.